app/_how-tos/enable-oidc-auth-for-dev-portal.md

```diff
@@ -0,0 +1,242 @@
+   * **Grant type:** Client Credentials
+   * **Scopes requested:** `api.access` 
+
+## Configure Okta OIDC application authentication in Dev Portal
+
+[Configure Okta OIDC application authentication](/api/konnect/application-auth-strategies/v2/#/operations/create-app-auth-strategy) using the `/v2/application-auth-strategies` endpoint:
+
+<!--vale off-->
+{% control_plane_request %}
+url: /v2/application-auth-strategies
+status_code: 201
+method: POST
+headers:
+    - 'Accept: application/json'
+    - 'Content-Type: application/json'
+    - 'Authorization: Bearer $DECK_KONNECT_TOKEN'
+body:
+    name: Okta OIDC
+    display_name: Okta OIDC
+    strategy_type: openid_connect
+    configs:
+        openid-connect:
+            issuer: $ISSUER_URL
+            credential_claim: 
+            - $OKTA_CLAIM_NAME
+            auth_methods: 
+            - client_credentials
+            scopes: 
+            - api.access
+{% endcontrol_plane_request %}
+<!--vale on-->
+
+Export the ID of the Okta OIDC auth strategy:
+
+```sh
+export AUTH_STRATEGY_ID='YOUR-AUTH-STRATEGY-ID'
+```
+
+## Apply the Okta OIDC auth strategy to an API
+
+Now that the application auth strategy is configured, you can [apply it to an API](/api/konnect/api-builder/v3/#/operations/publish-api-to-portal) using the `/v3/apis/{apiId}/publications/{portalId}` endpoint:
+
+<!--vale off-->
+{% control_plane_request %}
+url: /v3/apis/$API_ID/publications/$PORTAL_ID
+status_code: 201
+method: PUT
+headers:
+    - 'Accept: application/json'
+    - 'Content-Type: application/json'
+    - 'Authorization: Bearer $DECK_KONNECT_TOKEN'
+body:
+    auth_strategy_ids: 
+    - $AUTH_STRATEGY_ID
+{% endcontrol_plane_request %}
+<!--vale on-->
+
+This request will also publish the API to the specified Dev Portal.
+
+## Create an app in Dev Portal
+
+To use your Okta OIDC credentials to authenticate with an app, you must first create an app in Dev Portal with the [test developer account you created previously](/how-to/enable-oidc-auth-for-dev-portal/#create-a-dev-portal-developer-account).
+
+1. Navigate to your Dev Portal and log in with the test developer account:
+   ```sh
+   open https://$PORTAL_URL
+   ```
+   You should see `MyAPI` in the list of APIs.
+1. To register an app with the API, click **View APIs**.
+1. Click **Use this API**.
+1. In the pop-up dialog, enter a name for the app and your client ID for your Okta application.
+1. Click **Save**.
+
+## Validate the password grant
+
+Now, validate the setup by accessing the `example-route` Route and passing the user credentials in `Basic username:password` format. When you use the Okta OIDC authentication strategy, you use your Okta client ID as the username and your Okta client secret as the password, but they must be base64 encoded.
+
+First, encode your credentials and export them:
+```sh
+echo -n "$OKTA_CLIENT_ID:$OKTA_CLIENT_SECRET" | base64
+export ENCODED_CREDENTIALS='YOUR-ENCODED-CREDENTIALS'
+```
```

Comment:
Might be better to split these two so people don't run them together by accident. Some commands are fine to group, but you need the output of one to populate the other here.